### PR TITLE
Produce Source Distribution and Documentation as tar.gz

### PR DESCRIPTION
**Pull request by scbedd**
This PR updates our shippable source distributions from `zip` to `tar.gz`.

Not a huge change for customers, but needs a bunch of finicky changes throughout our scripting to accommodate.

Related to the first checkbox of #30192

### CODE REVIEW

File: eng/tox/tox_helper_tasks.py [hunk 1]
```diff
@@ -50,21 +44,37 @@ def get_pip_list_output():
...
+            return os.path.join(extract_location, extracted_dir)
+    else:
+        # .tar.gz -> .tar
+        with tarfile.open(path_to_zip_file) as tar_ref:
+            tar_ref.extractall(extract_location)
+            extracted_dir = os.path.basename(path_to_zip_file).replace(".tar.gz", "")
+            return os.path.join(extract_location, extracted_dir)
+
+
 def unzip_sdist_to_directory(containing_folder):
```
Comment (mccoyp) [inline eng/tox/tox_helper_tasks.py:72]: Did you mean to also remove this method? Currently `unzip_sdist_to_directory` is defined twice

Comment (scbedd) [reply eng/tox/tox_helper_tasks.py:72]: Yeah wtf haha

File: eng/tox/tox_helper_tasks.py [hunk 2]
```diff
@@ -50,19 +44,29 @@ def get_pip_list_output():
...
-    return unzip_file_to_directory(path_to_zip_file, containing_folder)
+def unzip_sdist_to_directory(containing_folder: str) -> str:
+    zips = glob.glob(os.path.join(containing_folder, "*.zip"))
+
+    if zips:
+        return unzip_file_to_directory(zips[0], containing_folder)
+    else:
+        tars = glob.glob(os.path.join(containing_folder, "*.tar.gz"))
+        val = unzip_file_to_directory(tars[0], containing_folder)
+        return val
```
Comment (mccoyp) [inline eng/tox/tox_helper_tasks.py:55]: nit:
```suggestion
        return unzip_file_to_directory(tars[0], containing_folder)
```

File: eng/tox/tox_helper_tasks.py [hunk 3]
```diff
@@ -50,19 +44,29 @@ def get_pip_list_output():
...
+def unzip_file_to_directory(path_to_zip_file: str, extract_location: str) -> str:
+    if path_to_zip_file.endswith(".zip"):
+        with zipfile.ZipFile(path_to_zip_file, "r") as zip_ref:
+            zip_ref.extractall(extract_location)
+            extracted_dir = os.path.basename(os.path.splitext(path_to_zip_file)[0])
+            return os.path.join(extract_location, extracted_dir)
+    else:
+        # .tar.gz -> .tar
+        with tarfile.open(path_to_zip_file) as tar_ref:
+            tar_ref.extractall(extract_location)
```
Comment (mccoyp) [inline eng/tox/tox_helper_tasks.py:67]: Does the `.tar.gz` extension change to `.tar` here?

Comment (scbedd) [reply eng/tox/tox_helper_tasks.py:67]: I don't believe so. I think I need to clean the comment.

**Review by mccoyp — APPROVED** [review]

**Comment by scbedd** [comment]
> ~@mccoyp This is ready to go. Tested attachment points + nightly.~
> 
> I discovered one last finicky issue.
> 
> image
> 
> And so the docs are going to the wrong place in github: https://azuresdkdocs.blob.core.windows.net/$web/python/azure-template.tar/0.1.0b2971272/index.html
> 
> Need to resolve before merging.

Resolved.

See [this new github.io release](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-template/0.1.0b2971647/index.html)

**Pull request merged**